### DISCUSSION TITLE
[Request] (auch: Bug) Olimex ETH ausschließlich über Ethernet betreiben

### DISCUSSION THREAD

**Discussion by Gerd-Watzmann** [tbnobody/OpenDTU#1197] [Q&A]
### Is your feature request related to a problem? Please describe.

Ich hatte mir das Olimex ETH Board (ESP32-POE-ISO) besorgt, in der Hoffnung damit eine DTU aufbauen zu können die sich ausschließlich über Ethernet ins Hausnetz einklinkt (WLAN steht nicht immer zur Verfügung).

Geflasht habe ich die aktuelle opendtu-generic.bin  23.8.1

Das Funkmodul habe ich erst einmal nicht angeschlossen.

Im Prinzip funktioniert dann erstmal alles ... im AP-Modus bekomme ich Kontakt zu openDTU, kann die Konfiguration vornehmen und eine feste IP-Adresse vergeben.

Aber: die Verbindung über Ethernet funktioniert nur so lange, wie eine gültige WLAN-Verbindung ins heimische Netz besteht. Es gelingt mir nicht, WLAN in den Einstellungen zu deaktivieren. Und wenn ich ungültige WLAN-Angaben mache, um eine ausschließliche Kommunikation über Ethernet zu erzwingen, geht auch der Kontakt über Ethernet verloren und ich muss die Firmware neu flashen.

Eine Frage von mir als Laie (leider kann ich nicht selber programmieren): 

Wäre es nicht möglich, eine Firmware so zu erstellen dass WLAN gar nicht erst benutzt wird? Also nur Ethernet aktivieren, IP-Adresse über DHCP beziehen und darüber die Web-Oberfläche zur weiteren Konfigauration zu Verfügung stellen?

Das wäre dann ganz einfach "straight forward", ohne irgendwelche komplizierten Umwege über WLAN und AP-Modus.

Oder stelle ich mir das zu einfach vor?

Sorry, falls ich zu dumm frage ;-) Ich finde es toll, vom Fachwissen dieser Community profitieren zu können. Danke Euch!

Ausgabe über die serielle Konsole beim Einschalten (Ethernet-Kabel ist eingesteckt, und Router sieht den Olimex auch):

Starting OpenDTU
Initialize FS... done
Reading configuration... done
Reading PinMapping... [   160][E][vfs_api.cpp:105] open(): /littlefs/pin_mapping.json does not exist, no permits for creation
using default config done
Initialize Network... done
Setting Hostname... Configuring WiFi STA using new credentials... done
Initialize NTP... done
Initialize SunPosition... done
Initialize MqTT... done
Initialize WebApi... done
Initialize Display... done
Initialize LEDs... done
Check for default DTU serial... done
Initialize Hoymiles interface... NRF: Connection error!!
  Setting radio PA level... 
  Setting DTU serial... 
  Setting poll interval... 
done
Switch to WiFi mode
Setting Hostname... done
Configuring WiFi STA using new credentials... done
Configuring WiFi STA static IP... done
WiFi disconnected
Try reconnecting
Network lost connection
WiFi disconnected
Try reconnecting
Network lost connection   (wird dann unendlich wiederholt)

### Describe the solution you'd like

Eine Firmware, welche Olimex ETH ausschließlich über Ethernet kommunizieren lässt.

**Comment by tbnobody** [comment]
Ich schiebe das Ganze mal in die Discussions da es sich hier eher um Support handelt statt um einen Issue oder wirkliches Enhancement. Aktuell kennt dein Gerät noch keine Pin Belegung für die Ethernet Hardware. Dafür musst du ein passendes Device Profile einspielen. https://github.com/tbnobody/OpenDTU/wiki/Device-Profiles

Für das Olimex Board gibt es sogar schon fertige Profile: https://github.com/tbnobody/OpenDTU/blob/master/docs/DeviceProfiles/olimex_esp32_poe.json

Einfach hochladen, aktivieren, geht.

**Reply by Gerd-Watzmann** [reply]
Jetzt geht es, mit ein wenig Rumprobieren :-)

Für mich als Laien war das jetzt nicht wirklich selbsterklärend - daher nochmals ganz herzlichen Dank für die Hilfe!

**Reply by Gerd-Watzmann** [reply]
Nachtrag: die Ethernet-Verbindung funktioniert jetzt einwandfrei.

Aber dafür habe ich jetzt den (auch schon von anderen Usern beschriebenen) Fehler, dass die Verbindung zu den Wechselrichtern immer nur für einige Minuten nach dem Sichern der Netzwerkeinstellungen vorhanden ist.

Nachdem ich mit AhoyDTU via Ethernet (W5500) schon Schiffbruch erlitten hatte und mir alle Welt zu OpenDTU geraten hat, komme ich da jetzt auch nicht weiter :-(   Irgendwie ist das Thema "DTU via Ethernet" wie verhext.
